src/Symfony/Component/DependencyInjection/Compiler/DefinitionErrorExceptionPass.php

```diff
@@ -33,6 +40,10 @@ protected function processValue($value, $isRoot = false)
         }
 
         if ($isRoot && !$value->isPublic()) {
+            if ($this->analyzingPass) {
+                $this->analyzingPass->process($this->container);
+                $this->analyzingPass = null;
```

Comment:
this means we analyze only once. So does it actually make sense to have it in `processValue` ?